src/blocks/VoiceApp.tsx

```diff
@@ -0,0 +1,305 @@
+import React, { useState, useRef, useEffect } from 'react';
+import { ComposerComponentProps, BlockModel, FeedComponentProps } from './types';
+import { Microphone } from '@mozartec/capacitor-microphone';
+import { FirebaseStorage } from '@capacitor-firebase/storage';
+import { nanoid } from 'nanoid';
+import Box from '@mui/material/Box';
+import Fab from '@mui/material/Fab';
+import MicIcon from '@mui/icons-material/Mic';
+import { CircularProgress } from '@mui/material';
+import { Capacitor } from '@capacitor/core';
+import { StopCircleRounded, PlayCircleRounded } from '@mui/icons-material';
+import { useRecordAudio } from './utils/RecordAudio';
+
+export const VoiceComposerComponent = ({ model, done }: ComposerComponentProps) => {
+  const [uploadProgress, setUploadProgress] = useState<number>(0);
+  const [uploading, setUploading] = useState(false);
+  const { startRecording, stopRecording, isRecording } = useRecordAudio();
+
+  const toggleRecord = async () => {
+    if (isRecording) {
+      try {
+        const recordedAudio = await stopRecording();
+        uploadToFirebase(recordedAudio);
+      } catch (error) {
+        console.error('Recording failed:', error);
+        alert('Recording failed');
+      }
+    } else {
+      await startRecording();
+    }
+  };
+
+  const uploadToFirebase = async (recordedAudio: { path?: string; webPath?: string; dataUrl: string }) => {
```

Comment:
can we move this to the microphone utils also? Every miniapp that uses the mic will also want to upload it too